evm/utils/fixture_tests.py

```diff
@@ -41,59 +53,79 @@ def _recursive_find_files(base_dir, pattern):
                 yield os.path.join(dirpath, filename)
 
 
+def find_fixture_files(fixtures_base_dir):
+    all_fixture_paths = _recursive_find_files(fixtures_base_dir, "*.json")
+    return all_fixture_paths
+
+
 @to_tuple
-def find_fixtures(fixtures_base_dir, normalize_fn, skip_fn=None, mark_fn=None, ignore_fn=None):
+def find_fixtures(fixtures_base_dir):
     """
     Helper function for JSON based fixture test suite.
 
     - `fixtures_base_dir`: the filesystem path under which JSON fixtures can be found.
     - `normalize_fn`: callback to normalize json fixture to internal format.
     - `skip_fn`: callback to skip any tests that should not be run.
     """
-    all_fixture_paths = _recursive_find_files(fixtures_base_dir, "*.json")
+    all_fixture_paths = find_fixture_files(fixtures_base_dir)
 
     for fixture_path in sorted(all_fixture_paths):
         with open(fixture_path) as fixture_file:
             fixtures = json.load(fixture_file)
 
-        for key in sorted(fixtures.keys()):
-            fixture_relpath = os.path.relpath(fixture_path, fixtures_base_dir)
+        for fixture_key in sorted(fixtures.keys()):
+            yield (fixture_path, fixture_key)
 
-            fixture_name = "{0}:{1}".format(fixture_relpath, key)
 
-            if ignore_fn:
-                if ignore_fn(fixture_relpath, key, fixtures[key]):
-                    continue
+@curry
+def filter_fixtures(all_fixtures, fixtures_base_dir, mark_fn=None, ignore_fn=None):
+    for fixture_data in all_fixtures:
+        fixture_path = fixture_data[0]
+        fixture_relpath = os.path.relpath(fixture_path, fixtures_base_dir)
 
-            normalized_fixture = normalize_fn(fixtures[key])
+        if ignore_fn:
+            if ignore_fn(fixture_relpath, *fixture_data[1:]):
```

Comment:
I'd find it clearer if we unpacked fixture_data into separate variables above (line 83) and then passed each of them here to ignore_fn(), otherwise it's not trivial to find out what arguments it expects. A docstring would also work